pkg/gateway/operations/headobject.go

```diff
@@ -43,10 +43,33 @@ func (controller *HeadObject) Handle(w http.ResponseWriter, req *http.Request, o
 		return
 	}
 
+	// range query
+	var rng httputil.Range
+	var rngErr error
+	// range query
+	rangeSpec := req.Header.Get("Range")
+	if len(rangeSpec) > 0 {
+		rng, rngErr = httputil.ParseRange(rangeSpec, entry.Size)
+		if rngErr != nil {
+			o.Log(req).WithError(err).WithField("range", rangeSpec).Debug("invalid range spec")
+			if errors.Is(rngErr, httputil.ErrUnsatisfiableRange) {
+				w.WriteHeader(http.StatusRequestedRangeNotSatisfiable)
```

Comment:
Why is the behaviour on HEAD different from the behaviour on GET?  I didn't check the spec, but [MDN says](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD) it should send the exact same headers -- why not the same response?

("Because S3" is a good answer!)

Reply:
Because S3 :)
In case of GetObject, the returned response would also be a 416 but with an XML body describing the error. For HeadObject it's simply a 416

Reply:
So not "because S3" but "because S3 and XML" 🤦🏽.  Now I'm sorry I ever asked this.